pkg/apis/workflow/v1alpha1/event_types.go

```diff
@@ -0,0 +1,38 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WorkflowEvent is the definition of an event resource
+// +genclient
+// +genclient:noStatus
+// +kubebuilder:resource:shortName=wfev
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type WorkflowEvent struct {
```

Comment:
I wonder if, because this defines (a) an event matcher and (b) what to do when that event is matched maybe call it something like `WorkflowEventAction` or `wea`